src/raster/r.viewshed.exposure/r.viewshed.exposure.py

```diff
@@ -0,0 +1,1391 @@
+    dsm_type = grass.parse_command("r.info", map=r_dsm, flags="g")["datatype"]
+
+    # Test if DSM exist
+    gfile_dsm = grass.find_file(name=r_dsm, element="cell")
+    if not gfile_dsm["file"]:
+        grass.fatal("Raster map <{}> not found".format(r_dsm))
+
+    # Exposure settings
+    v_source = options["sampling_points"]
+    r_source = options["source"]
+    source_cat = options["sourcecat"]
+    r_weights = options["weights"]
+
+    # test if source vector map exist and contains points
+    if v_source:
+        gfile_vsource = grass.find_file(name=v_source, element="vector")
+        if not gfile_vsource["file"]:
+            grass.fatal("Vector map <{}> not found".format(v_source))
+        if not grass.vector.vector_info_topo(v_source, layer=1)["points"] > 0:
+            grass.fatal("Vector map <{}> does not contain any points.".format(v_source))
+
+    if r_source:
+        gfile_rsource = grass.find_file(name=r_source, element="cell")
+        if not gfile_rsource["file"]:
+            grass.fatal("Raster map <{}> not found".format(r_source))
+
+        # if source_cat is set, check that r_source is CELL
+        source_datatype = grass.parse_command("r.info", map=r_source, flags="g")[
+            "datatype"
+        ]
+
+        if source_cat != "*" and source_datatype != "CELL":
+            grass.fatal(
+                "The raster map <%s> must be integer (CELL type) in order to \
+                use the 'sourcecat' parameter"
+                % r_source
+            )
+
+    if r_weights:
+        gfile_weights = grass.find_file(name=r_weights, element="cell")
+        if not gfile_weights["file"]:
+            grass.fatal("Raster map <{}> not found".format(r_weights))
+
+    # Viewshed settings
+    range_inp = float(options["range"])
+    v_elevation = float(options["observer_elevation"])
+    b_1 = float(options["b1_distance"])
+    pfunction = options["function"]
+    refr_coeff = float(options["refraction_coeff"])
+    flagstring = ""
+    if flags["r"]:
+        flagstring += "r"
+    if flags["c"]:
+        flagstring += "c"
+
+    # test values
+    if v_elevation < 0.0:
+        grass.fatal("Observer elevation must be larger than or equal to 0.0.")
+
+    if range_inp <= 0.0 and range_inp != -1:
+        grass.fatal("Exposure range must be larger than 0.0.")
+
+    if pfunction == "Fuzzy_viewshed" and range_inp == -1:
+        grass.fatal(
+            "Exposure range cannot be \
+            infinity for fuzzy viewshed approch."
+        )
+
+    if pfunction == "Fuzzy_viewshed" and b_1 > range_inp:
+        grass.fatal(
+            "Exposure range must be larger than radius around \
+            the viewpoint where clarity is perfect."
+        )
+
+    # Sampling settings
+    source_sample_density = float(options["sample_density"])
+    seed = options["seed"]
+
+    if not seed:  # if seed is not set, set it to process number
+        seed = os.getpid()
+
+    # Optional
+    cores = int(options["cores"])
```

Comment:
The should be nprocs now, also needs to be changed in manual.

Reply:
Thanks. Fixed in 4b78c6e.